src/cart/internal-line-item.ts

```diff
@@ -13,6 +13,7 @@ export default interface InternalLineItem {
     quantity: number;
     type: string;
     variantId: number | null;
+    productId?: number;
```

Comment:
I believe it should be `productId` (without the `?`). If you look at `LineItem`, you can see that `productId` is not an optional property.

Reply:
When I added it as a `mandatory` property, it throws error for `GiftCertificateItem` since its returning  the `InternalLineItem` object @ https://github.com/bigcommerce/checkout-sdk-js/blob/b14beeea0456e1fb7c39cd445faef3161a84afed/src/cart/map-gift-certificate-to-internal-line-item.ts#L9. 

So I have discussed it with @lord2800 and we decided to add it as a optional property.

Reply:
Ah I see, thanks for explaining. I'm okay with this.

Reply:
squashed the commits.